water/include/water/Dialect/Wave/IR/WaveUtils.h

```diff
@@ -40,7 +40,7 @@ getUncollapsedVectorShape(llvm::ArrayRef<wave::WaveSymbolAttr> shape,
 /// Resolve named Wave symbols to concrete integer values using the
 /// hyperparameter table.
 std::optional<llvm::SmallVector<int64_t>>
-resolveSymbolNames(llvm::ArrayRef<wave::WaveSymbolAttr> names,
+resolveSymbolNames(llvm::ArrayRef<llvm::StringRef> names,
```

Comment:
This should normally remain as `WaveSymbolAttr`, special IDs never appear in hyperparameters, we never fix the thread identifier there.